apps/backend/src/index.d.ts

```diff
@@ -0,0 +1,11 @@
+import { type FastifyRequest } from "fastify";
+
+import { type UserAuthResponseDto } from "./modules/users/users.js";
+
+type OriginalFastifyRequest = FastifyRequest;
+
+declare module "fastify" {
+	interface FastifyRequest extends OriginalFastifyRequest {
+		user: null | UserAuthResponseDto;
+	}
```

Comment:
```suggestion
declare module "fastify" {
	interface FastifyRequest {
		user: null | UserAuthResponseDto;
	}
```
you can declare the interface with the same name and the typescript will merge them automatically